conjure-java-core/src/main/java/com/palantir/conjure/java/util/JavaNameSanitizer.java

```diff
@@ -30,11 +37,48 @@
      */
     public static String sanitize(FieldName fieldName) {
         String identifier = CaseConverter.toCase(fieldName.get(), CaseConverter.Case.LOWER_CAMEL_CASE);
-        return sanitize(identifier);
+        return sanitizeFieldName(identifier);
     }
 
-    private static String sanitize(String name) {
-        return SourceVersion.isName(name) && !RESERVED_FIELD_NAMES.contains(name) ? name : name + "_";
+    /** Returns a valid java name based on the input. */
+    public static String sanitize(String name) {
+        return SourceVersion.isName(name) ? name : escape(name);
+    }
+
+    /** Sanitizes parameter names taking into account default auth type parameter names. */
+    public static String sanitizeParameterName(String input, EndpointDefinition endpoint) {
+        String value = sanitize(input);
+        Optional<String> maybeAuthParamName = endpoint.getAuth()
```

Comment:
Same as above, I would just add authHeader and cookieToken to the list of reserved keywords.

Reply:
I've seen endpoints which use `auth: none` and define an `authHeader` header parameters, so I'd prefer not to mangle those unnecessarily.
Originally I planned to add these and `result` to the restricted list, but this example made me reconsider.

Reply:
That's fair but tbh I don't really see much downside of over-sanitizing.